services/QuillLMS/app/services/quill_big_query/runner.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# All messages to BigQuery should use this class
+
+module QuillBigQuery
+  class Runner
+    PROJECT_ID = 'analytics-data-stores'
+
+    class UnsupportedSchemaError < StandardError; end
+    class ClientExecutionError < StandardError; end
+
+    def self.valid_schema?(json_body)
+      json_body['schema']['fields'].respond_to?(:count) &&
+        json_body['rows'].respond_to?(:count)
+    end
+
+    def self.transform_response(json_body)
+      raise UnsupportedSchemaError, json_body unless valid_schema?(json_body)
+
+      # This parses to [{"id" => 1, "name"=> "Dan", }, {"id" => 2, "name" => "Peter"}]
+      raw_fields = json_body["schema"]["fields"]
+      fields = raw_fields.pluck("name")
+      values = json_body['rows'].pluck('f').map { |val_hash| val_hash.pluck('v') }
+      hash_results = values.map {|v| fields.zip(v).to_h}
+      QuillBigQuery::Transformer.new(raw_fields, hash_results).transform
```

Comment:
Transformer seems like a good candidate for ApplicationService:

QuillBigQuery::Transformer.run(raw_fields, hash_results)

Reply:
Hmm, it would certainly work here. I tend to prefer PORO's over the ApplicationService pattern, but this seems like a good thing to discuss in the next engineering meeting, since it would behoove us all to align on the criteria for deciding on when to use this pattern.

Reply:
I went with your suggestion to use ApplicationService for the client in this PR, since that felt like more of a "service" to me. But I realize my definition of "service" is not tightly defined.